### PR TITLE
Add non-root support in sysvinit script

### PR DESCRIPTION
**Pull request by dreampuf**
As #4333 mentioned, I found that the sysvinit script of beats will fail on non-root user launch the beats. Cause [the arguments checker](https://github.com/elastic/beats/blob/c10d47af73f51b3db2a46ffb606e771bb8d397b3/libbeat/common/config.go#L432) will be executed as root by default. 
I add an optional `BEAT_USER`, which may come from `/etc/default/$NAME`.

### CODE REVIEW

**Comment by andrewkroh** [comment]
jenkins, package it

**Comment by andrewkroh** [comment]
What provides `runuser`? And it is safe to assume that it is present?

**Comment by dreampuf** [comment]
@andrewkroh `runuser`([man page](http://man7.org/linux/man-pages/man1/runuser.1.html)) is a tool released by `util-linux-2.23`[link](https://www.kernel.org/pub/linux/utils/util-linux/v2.23/v2.23-ReleaseNotes) . It has been provided in most popular distributions. CentOS 6+, Debian 8+, Ubuntu 16.04+.
`su` is another option, there are some comments of these two. Details: http://danwalsh.livejournal.com/55588.html

Basically, if Filebeat plan to support some older distribution. CentOS5/Debian7/Ubuntu14.04. `su USER -c` will be the option. Otherwise `runuser` is a better choice.

**Comment by andrewkroh** [comment]
Thanks for the details on `runuser` vs `su`. We do need to support ubuntu 14.04 and debian 7. Could it test if `runuser` is available and fallback to `su` if not?

**Comment by andrewkroh** [comment]
Can you please sign the CLA (contributor license agreement)? http://www.elasticsearch.org/contributor-agreement/

**Comment by dreampuf** [comment]
@andrewkroh `su` added. Also signed CLA.

File: dev-tools/packer/platforms/centos/init.j2 [hunk 1]
```diff
@@ -42,8 +44,12 @@ if status | grep -q -- '-p' 2>/dev/null; then
     pidopts="-p $pidfile"
 fi
 
+if [ -f /sbin/runuser ]; then
```
Comment (andrewkroh) [inline dev-tools/packer/platforms/centos/init.j2]: Rather than explicitly stating the path, I think it would be better to use the check described [here](http://stackoverflow.com/a/4785518/503798). WDYT?

File: dev-tools/packer/platforms/centos/init.j2 [hunk 2]
```diff
@@ -42,8 +44,12 @@ if status | grep -q -- '-p' 2>/dev/null; then
...
+    user_wrapper="runuser"
```
Comment (andrewkroh) [inline dev-tools/packer/platforms/centos/init.j2]: Do you also need to update to `user_wrapperopts="-u $beat_user"`?

Comment (dreampuf) [inline dev-tools/packer/platforms/centos/init.j2]: It's same argument and order. So I can reuse `user_wrapperopts` here.

File: dev-tools/packer/platforms/centos/init.j2 (hunk 1 above)
Comment (dreampuf) [inline dev-tools/packer/platforms/centos/init.j2]: Good point.

**Comment by dreampuf** [comment]
@andrewkroh It seems like only a CLA need to be done from this checklist. But I had signed a CLA yesterday(with my github username and email address). Anything I can do?

**Comment by tsg** [comment]
jenkins, test it

**Comment by tsg** [comment]
We should have a CHANGELOG entry on this one. @dreampuf could you add a line under the `Added` section, please? I can also easily add it in a follow up PR if you prefer to merge it quickly.

**Comment by tsg** [comment]
Jenkins test failure is not related.
